tsfresh/feature_extraction/extraction.py

```diff
@@ -172,29 +176,35 @@ def _extract_features_for_one_time_series(prefix_and_dataframe, column_id, colum
     column_prefix, dataframe = prefix_and_dataframe
     column_prefix = str(column_prefix)
 
-    if settings.set_default and column_prefix not in settings.kind_to_calculation_settings_mapping:
-        settings.set_default_parameters(column_prefix)
+    with warnings.catch_warnings():
+        if not settings.show_warnings:
+            warnings.simplefilter("ignore")
+        else:
+            warnings.simplefilter("default")
+
+        if settings.set_default and column_prefix not in settings.kind_to_calculation_settings_mapping:
+            settings.set_default_parameters(column_prefix)
 
-    # Calculate the aggregation functions
-    column_name_to_aggregate_function = settings.get_aggregate_functions(column_prefix)
+        # Calculate the aggregation functions
```

Comment:
Ohm, these are whitespaces, but I have to increase the indention, because this is all in a context wrapper... or am I getting something wrong here? :-)